lib/checksum.h

```diff
@@ -32,11 +32,15 @@ struct ipv6_ph {
 
 extern uint16_t in_cksumv(const struct iovec *iov, size_t iov_len);
```

Comment:
just looking at this, all the casting seems due to the use of iovec in this base api. but we only have one or two inputs (no header or with header), so ... maybe we should just not use iovec and use a pair of pointers-and-lengths in the base api?

Reply:
It's the `void` that's causing the problem here :laughing: … the compiler magic breaks because you can't declare a variable of type `void` and the function trick doesn't work in C++…

Reply:
maybe I don't understand the problem?
if we offered the base api with (const uint8_t *ptr1, int len1, const uint8_t *ptr2, int len2), couldn't we do what we need to do, without the casting and iovec?

Reply:
I'm not exactly disagreeing, just saying the problem here isn't `iovec`, it's the `const void *` input pointers. If those were `const char *` or `const uint8_t *` everything is fine, even if `iovec` is still used.

Reply:
The `const void *` input on `in_cksum` / `in_cksum_with_ph4` / `in_cksum_with_ph6` I mean.

Reply:
sorry, sort of get it now - I see there are just a handful of users, maybe I can tweak things to make myself happier, but no need to involve this PR in that...